src/pages/goodbye-clean-code/index.pt-br.md

```diff
@@ -0,0 +1,182 @@
+
+O código parecia um pouco com isso:
+
+```jsx
+let Retangulo = {
+  redimensionarTopoEsquerdo(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarTopoDireito(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarBaseEsquerda(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarBaseDireita(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+};
+
+let Oval = {
+  redimensionarEsquerda(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarDireita(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarTopo(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarBase(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+};
+
+let Cabecalho = {
+  redimensionarEsquerda(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarDireita(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+}
+
+let BlocoDeTexto = {
+  redimensionarTopoEsquerda(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarTopoDireito(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarBaseEsquerda(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+  redimensionarBaseDireita(posicao, tamanho, preservarProporcao, dx, dy) {
+    // 10 linhas repetitivas de matemática
+  },
+};
+```
+
+Aquela matemática repetitiva realmente me incomodava.
+
+Não era _limpo_.
+
+A maior parte da repetição era entre as direções semelhantes. Por exemplo, `Oval.redimensionarEsquerda()` tinha semelhanças com `Cabecalho.redimensionarEsquerda()`. Isso era porque ambas lidavam com arrastar a alça no lado esquerdo.
+
+A outra semelhança era entre os métodos para a mesma forma. Por exemplo, `Oval.redimensionarEsquerda()` tinha semelhanças com os outros métodos de `Oval`. Isso acontecia porque todos lidavam com formas ovais. Também havia duplicação entre `Retangulo`, `Cabecalho` e `BlocoDeTexto` porque todos *eram* retângulos.
+
+Eu tive uma ideia.
+
+Poderíamos *remover todas as duplicações* agrupando o código dessa maneira:
+
+```jsx
+let Direcoes = {
+  topo(...) {
```

Comment:
Seems a bit weird to translate the code? Maybe let's keep it as is?

Reply:
Ok, no problem with that. Although I'd ask what do you think about keeping the comments translated, @gaearon?